PetaPoco.Tests.Unit/Utilities/PagingHelperTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text;
+using PetaPoco.Utilities;
+using Shouldly;
+using Xunit;
+
+namespace PetaPoco.Tests.Unit.Utilities
+{
+    public class PagingHelperTests
+    {
+        [Fact]
+        public void SplitSQL_GivenBigSqlStringWithOrWithoutOrderBy_ShouldPerformTheSameInBothCases()
+        {
+            var inputWithOrderBy = GenerateSql(200, 5, 100, true);
+            var inputWithoutOrderBy = inputWithOrderBy.Substring(0, inputWithOrderBy.IndexOf("ORDER BY"));
+
+            var sw = Stopwatch.StartNew();
+
+            PagingHelper.Instance.SplitSQL(inputWithOrderBy, out SQLParts partsWithOrderBy);
+            sw.Stop();
+            var elapsedMillSecondsWithOrderBy = sw.ElapsedMilliseconds;
+
+            sw.Restart();
+
+            PagingHelper.Instance.SplitSQL(inputWithoutOrderBy, out SQLParts partsWithoutOrderBy);
+            sw.Stop();
+            var elapsedMillSecondsWithoutOrderBy = sw.ElapsedMilliseconds;
+
+            var actualTimeDiffInMilliSeconds = Math.Abs(elapsedMillSecondsWithOrderBy - elapsedMillSecondsWithoutOrderBy);
+            var maxExpectedTimeDiffInMilliSeconds = 200;
+            actualTimeDiffInMilliSeconds.ShouldBeLessThanOrEqualTo(maxExpectedTimeDiffInMilliSeconds,
+                $"{nameof(PagingHelper.SplitSQL)} degrades in performance. Expected {maxExpectedTimeDiffInMilliSeconds}ms or less, but was {actualTimeDiffInMilliSeconds}ms.");
+        }
+
```

Comment:
GitHub's online diffs are pretty worthless, so in case you're viewing this in the online interface, this might help:

![1698087837-●_PetaPoco_-_PetaPoco Tests UnitUtilitiesPagingHel](https://github.com/CollaboratingPlatypus/PetaPoco/assets/37424493/33eba997-7768-4137-a4fb-a075e9bdad50)

Reply:
I know. Had to open my laptop and use the "big screen" to avoid crossed eyes lol